### PR TITLE
User/elormcoch/wsl extension fix

### PR DESCRIPTION
**Pull request by ElormCoch**
Adapts the launch and debug commands based on whether the extension is run in a remote environment.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -239,20 +239,37 @@ export function activate(context: vscode.ExtensionContext): void {
...
         const devToolsAttachConfig = providedLaunchDevToolsConfig;
-        edgeDebugConfig.url = `file://${fileUri.fsPath}`;
-        devToolsAttachConfig.url = `file://${fileUri.fsPath}`;
-        void vscode.debug.startDebugging(undefined, edgeDebugConfig).then(() => vscode.debug.startDebugging(undefined, devToolsAttachConfig));
+        if (typeof vscode.env.remoteName === 'undefined') {
+            edgeDebugConfig.url = `file://${fileUri.fsPath}`;
+            devToolsAttachConfig.url = `file://${fileUri.fsPath}`;
+            void vscode.debug.startDebugging(undefined, edgeDebugConfig).then(() => vscode.debug.startDebugging(undefined, devToolsAttachConfig));
+        } else {
+            const url = `file://${vscode.env.remoteName}.localhost/${fileUri.authority.split('+')[1]}/${fileUri.fsPath.replace(/\\/g, '/')}`;
```
Comment (vidorteg) [inline src/extension.ts:251]: could you add a comment with an example of the expected shape of this url?

File: src/extension.ts [hunk 2]
```diff
@@ -239,20 +239,37 @@ export function activate(context: vscode.ExtensionContext): void {
         void setCSSMirrorContentEnabled(context, !cssMirrorContent);
     }));
 
-    context.subscriptions.push(vscode.commands.registerCommand(`${SETTINGS_VIEW_NAME}.launchHtml`, (fileUri: vscode.Uri): void => {
+    context.subscriptions.push(vscode.commands.registerCommand(`${SETTINGS_VIEW_NAME}.launchHtml`, async (fileUri: vscode.Uri): Promise<void> => {
```
Comment (vidorteg) [inline src/extension.ts:242]: could you also add a test for this scenario in:

https://github.com/microsoft/vscode-edge-devtools/blob/main/test/extension.test.ts

that way we will avoid future regressions.

**Review by vidorteg — COMMENTED** [review]
Thanks for the fix 👏 @ElormCoch, most of code LGTM,  I left a couple of comments, could you PTAL?

**Review by HOLYGHOST1111 — APPROVED** [review]

File: src/extension.ts (hunk 2 above)
Comment (ElormCoch) [reply src/extension.ts:242]: Done. Thanks!

**Review by vidorteg — APPROVED** [review]
Couple of nits but aside from that LGTM 🥳

**Comment by ElormCoch** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by vidorteg** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**